### ISSUE TITLE
Enabled Filters on subresource

### ISSUE THREAD

**Issue by lvillarino** [api-platform/api-platform#634]
Hi,

I want to know how I can enabled a filter to a Subresource?
Ex, We have two entities answers and comments and I want to filter all the comments that contains a keyword when I go a GET request to answers/{id}/comments

Thanks

**Comment by soyuka** [comment]
It should work like other filters. What did you try?

**Comment by Simperfit** [comment]
@lvillarino could you please check what @soyuka said ?

**Comment by alinepure** [comment]
@lvillarino, did you manage to make it work? I try to activate the filter for a subresource too and it doesn't work.

I have two entities: Sheet and Record. Sheet has an @ApiSubresource on Record which returns a collection.
I've set up an exists filter on the end class (Record entity).

The filter is not working on **GET /sheets/{id}/records?parentRecord[exists]=true**, but working well on **GET /records?parentRecord[exists]=true**

**Comment by alinepure** [comment]
@soyuka, the [resourceFilters](https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/Orm/Extension/FilterExtension.php#L61) are empty for subresource. Should we add the filters property under the subresourceOperation too?

**Comment by soyuka** [comment]
> Should we add the filters property under the subresource operation too?

Yes, or just use the ApiFilter annotation is should work seemlessly.

**Comment by alinepure** [comment]
Weird, by setting up the filters on subresource operation doesn't work, but it works with the annotation. Thanks!

**Comment by desmax** [comment]
I want to add filter specifically to subresource operation and not globally per resource. this does not work

**Comment by konandrum** [comment]
> setting up the filters on subresource operation doesn't work, but it works with the

Get the same trouble ...

**Comment by lucasgranberg** [comment]
The problem is here: https://github.com/api-platform/core/blob/3a8329f09ded4d42e807ebe92006921efebbfe98/src/OpenApi/Factory/OpenApiFactory.php#L374

getFiltersParameter always looks in the collectionOperations for the filter. To fix it for me I changed the beginning of the method like this:
```php
    private function getFiltersParameters(ResourceMetadata $resourceMetadata, string $operationName, string $resourceClass): array
    {
        $parameters = [];
        if(isset($resourceMetadata->getCollectionOperations()[$operationName])){
            $resourceFilters = $resourceMetadata->getCollectionOperationAttribute($operationName, 'filters', [], true);
        }else if(isset($resourceMetadata->getSubresourceOperations()[$operationName])){
            $resourceFilters = $resourceMetadata->getSubresourceOperationAttribute($operationName, 'filters', [], true);
        }
        ...
   }
```

**Comment by tonykiselis** [comment]
> ```
> getFiltersParameters
> ```

Yeah. That should work.
However, all scoping is private.
How do you override the factory class?
But at this point, it seems we have to override the entire class.

**Comment by lucasgranberg** [comment]
final classes are never final when you find out they are final.. I had to override the whole class and copy stuff.

**Comment by MrG0d** [comment]
Filters are enabled when I duplicate the operation in 'collectionOperations' in the entity config which is a sub-resource 
This allowed not to change the source of the ApiPlatform as [here](https://github.com/api-platform/api-platform/issues/634#issuecomment-788976096)
user.yaml
```
App\Entity\User:
  subresourceOperations:
    api_users_tickets_get_subresource:
      security: 'object == user'
```

ticket.yaml:
```
App\Entity\Ticket:
  collectionOperations:
    api_users_tickets_get_subresource:
      filters:
          - app.filter.ticket_by_user.search
```

**Comment by soyuka** [comment]
Closing per #2987. Questions are best asked on [GitHub Discussions](https://github.com/api-platform/core/discussions) or [Stack Overflow (api-platform tag)](https://stackoverflow.com/questions/tagged/api-platform.com). The framework now lives at `api-platform/core`.
